### PR TITLE
Add unverified DTO conversion methods

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: types/src/block/output/alias.rs [hunk 1]
```diff
@@ -753,6 +765,44 @@ pub mod dto {
...
+                value
+                    .amount
+                    .parse::<u64>()
+                    .map_err(|_| DtoError::InvalidField("amount"))?,
+                (&value.alias_id).try_into()?,
+            )?;
+
+            builder = builder.with_state_index(value.state_index);
+
+            if !value.state_metadata.is_empty() {
```
Comment (Thoralf-M) [inline types/src/block/output/alias.rs:779]: Why check for not empty?

File: types/src/block/output/alias.rs [hunk 2]
```diff
@@ -204,7 +204,7 @@ impl AliasOutputBuilder {
     }
 
     ///
-    pub fn finish(self, token_supply: u64) -> Result<AliasOutput, Error> {
+    fn _finish(self) -> Result<AliasOutput, Error> {
```
Comment (Thoralf-M) [inline types/src/block/output/alias.rs]: Just use `finish_unverified` and remove the current `finish_unverified`

Comment (thibault-martinez) [reply types/src/block/output/alias.rs]: Yes sir!

File: types/src/block/output/unlock_condition/mod.rs [hunk 3]
```diff
@@ -626,6 +626,62 @@ pub mod dto {
...
+                    let address: Address = (&v.address)
+                        .try_into()
+                        .map_err(|_e| DtoError::InvalidField("ImmutableAliasAddressUnlockCondition"))?;
+                    // An ImmutableAliasAddressUnlockCondition must have an AliasAddress.
+                    if let Address::Alias(alias_address) = &address {
+                        UnlockCondition::ImmutableAliasAddress(ImmutableAliasAddressUnlockCondition::new(
+                            *alias_address,
+                        ))
+                    } else {
+                        return Err(DtoError::InvalidField("ImmutableAliasAddressUnlockCondition"));
```
Comment (Thoralf-M) [inline types/src/block/output/unlock_condition/mod.rs]: Errors should be camelCase

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**